### PR TITLE
[Issue #83] new app structure

### LINKED ISSUE #83: Fix page structure so it is more mobile friendly
# Description
Create a component that has a fixed top navigation bar with scrollable content underneath. 

# Definition of Done
The top navigation bar should have an arrow to go back to the previous screen, a title and a burger button.

### PR DESCRIPTION
**Pull request by samueldominguez**
## Description

Abstract out the page concept with new component `AppWrapper` and wrap all pages with it.

## Changes

- Implemented AppWrapper: responsive template with configurable top navigation bar, scrollable content underneath and removed bottom navigation bar (those links will now go in the burger menu).
- Fixed CSS according to new wrapping component and changes in button placement (they were at the bottom of the screen, now they are just after the form inputs, this is due to the mobile to web pivot and the way the bottom part of the viewport can be hidden on mobile).

## Checks

- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

Should close #83

### CODE REVIEW

File: src/components/molecules/AppPageHeader.tsx [hunk 1]
```diff
@@ -1,14 +1,13 @@
-import AppBackButton from "../atom/AppBackButton";
-import PageHeaderProps from "../../types/PageHeaderProps";
+interface PageHeaderProps {
+    title?: string | null;
```
Comment (yodra) [inline src/components/molecules/AppPageHeader.tsx:2]: Why do we need to define a `null` value? title and description could be undefined is it really necessary to take into account `null` value? Also, does make sense title and description are optional? this component just prints the title and description I think at least one of them should be required

File: src/components/molecules/AppWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    titleClassName?: string,
+    showBurger?: boolean
+    children?: ReactNode
+}
+
+const AppWrapper = ({link, title, titleClassName = '', showBurger = false, children}: AppWrapperProps): JSX.Element => {
+    return (
+        <div className="bg-page text-app-black h-screen">
+            <div className="flex flex-col justify-center mx-auto w-full md:w-1/2 lg:w-1/3 px-8 pt-8">
+                {(showBurger || link || title) &&
```
Comment (yodra) [inline src/components/molecules/AppWrapper.tsx:18]: I think we could simplify this a bit more. We are checking if showBurger, link, or title are defined to print one of them, but also in lines 21 and 30 we are checking again if showBurger, or link are defined.

To understand more the context, is the title always printed? 
If you want we could have a quick call to discuss it 😉

**Review by yodra — COMMENTED** [review]
LGTM! I just added 2 comments, if you want we could have a call to discuss it 🙂

File: src/components/molecules/AppPageHeader.tsx (hunk 1 above)
Comment (samueldominguez) [reply src/components/molecules/AppPageHeader.tsx:2]: @yodra The reason for the null is that usually these props are fed from a `translate(foo.bar)` call which retrieves the string according to the language being used, and the function can return a `null` value. To save a lot of typing in a lot of places I decided to allow the null value. What do you think?

**Review by yodra — APPROVED** [review]

**Review by Saixel — APPROVED** [review]

**Comment by yodra** [comment]
Approved with the condition to improve the [AppPageHeader.tsx](https://github.com/TheTributeCommunity/fesbal-frontend/pull/84/files/ad9f6195b2c1775e79a29f21de97307b0121d351#diff-4694edc615725f4696790f106b3770c5e7c54cd061ad9c9f90bb5587033ff781) component to remove null types

**Review by rdiaz82 — APPROVED** [review]
LGTM

**Pull request merged**